bundles/org.openhab.core.persistence/src/main/java/org/openhab/core/persistence/extensions/PersistenceExtensions.java

```diff
@@ -476,15 +477,21 @@ private static void internalPersist(Item item, TimeSeries timeSeries, @Nullable
                         if (!forward && !historicItem.getState().equals(state)) {
                             // Last persisted state value different from current state value, so it must have updated
                             // since last persist. We do not know when from persistence, so get it from the item.
-                            return item.getLastStateUpdate();
+                            return Optional.ofNullable(item.getLastStateUpdate())
+                                    .map(instant -> instant.atZone(ZoneId.systemDefault())).orElse(null);
```

Comment:
The class already has a reference to `TimeZoneProvider`, so I think the converion should be done using the configured timezone.

Reply:
done